docs: TraceConfig and context attributes (#793)
Arize-ai/openinference: python/openinference-instrumentation/README.md

```shell
pip install openinference-instrumentation
```

## Customizing Spans

This core `openinference-instrumentation` package offers convenient tools to read & write from the OTEL Context. The OTEL context is then used to propagate attributes to every span created. We do this using Python context managers:

> * `using_session`: to specifiy a session ID.

Reviewer: ```suggestion
* `using_session`: to specifiy a session ID to track multi-turn conversations with a user
```

* `using_user`: to specifiy a user ID.
* `using_metadata`: to add custom metadata.
* `using_tag`: to add tags.
* `using_prompt_template`: to reflect the prompt template used.

> * `using_attributes`: to specify all the above at once.

Reviewer: give the rationale as to why you would want to set these. Readers might not know why or how they might want to use these

Author: Done

  
For example:
```python
from openinference.instrumentation import using_attributes

> tags = ["tag_1", "tag_2", ...]
> metadata = {
>     "key-1": value_1,
>     "key-2": value_2,

Reviewer: use real-life examples - it's more powerful
```suggestion
tags = ["business_critical", "topic: weather", ...]
metadata = {
    "country": "United States",
```

    ...
}
prompt_template = "Please describe the weather forecast for {city} on {date}"
prompt_template_variables = {"city": "Johannesburg", "date":"July 11"}
prompt_template_version = "v1.0"
with using_attributes(
    session_id="my-session-id",
    user_id="my-user-id",
    metadata=metadata,
    tags=tags,
    prompt_template=prompt_template,
    prompt_template_version=prompt_template_version,
    prompt_template_variables=prompt_template_variables,
):
    # Calls within this block will generate spans with the attributes:
    # "session.id" = "my-session-id"
    # "user.id" = "my-user-id"
    # "metadata" = "{\"key-1\": value_1, \"key-2\": value_2, ... }" # JSON serialized
    # "tag.tags" = "["tag_1","tag_2",...]"
    # "llm.prompt_template.template" = "Please describe the weather forecast for {city} on {date}"
    # "llm.prompt_template.variables" = "{\"city\": \"Johannesburg\", \"date\": \"July 11\"}" # JSON serialized
    # "llm.prompt_template.version " = "v1.0"
    ...
```
You can read more about this in our [docs](https://docs.arize.com/phoenix/tracing/how-to-tracing/customize-spans).

## Tracing Configuration

> The core `openinference-instrumentation` package offers the `TraceConfig` class, which helps set up privacy configurations to mask specific fields in your spans. For instance, you may want to keep sensitive information from being logged for security reasons, or you may want to limit the size of the base64 encoded images logged to reduced payload size.

Reviewer: ```suggestion
This package contains the central `TraceConfig` class, which lets you specify a tracing configuration that let's you control things like data privacy and payload sizes. For instance, you may want to keep sensitive information from being logged for security reasons, or you may want to limit the size of the base64 encoded images logged to reduced payload size.
```
